src/fread.c

```diff
@@ -1009,27 +1003,34 @@ static void parse_iso8601_timestamp(FieldParseContext *ctx)
   int32_t date, hour=0, minute=0, tz_hour=0, tz_minute=0;
   double second=0;
 
+  *target = NA_FLOAT64;
+
   parse_iso8601_date_core(&ch, &date);
   if (date == NA_INT32)
-    goto fail;
+    return;
   if (*ch != ' ' && *ch != 'T')
-    goto date_only;
-    // allows date-only field in a column with UTC-marked datetimes to be parsed as UTC too; test 2150.13
+  {
+    *target = 86400 * (double)date;
+    
+    *(ctx->ch) = ch;
+    return;
+  }
+  // allows date-only field in a column with UTC-marked datetimes to be parsed as UTC too; test 2150.13
   ch++;
 
   str_to_i32_core(&ch, &hour, true);
   if (hour == NA_INT32 || hour < 0 || hour > 23 || *ch != ':')
-    goto fail;
+    return;
```

Comment:
looks like these are previously uncovered too:

https://app.codecov.io/gh/Rdatatable/data.table/blob/master/src%2Ffread.c#L1022

Would you mind adding some unit tests while we're here?

Reply:
This is more of a boyscout rule request:

https://iamjosephmj.medium.com/the-boy-scout-rule-leaving-code-better-than-you-found-it-fc21857ef8a1